### DISCUSSION TITLE
I2S-webserver_wav sound is very very quiet

### DISCUSSION THREAD

**Discussion by wimverlindensa** [pschatzmann/arduino-audio-tools#187] [Q&A] [answered]
The soundlevel is very very low, have to open volume of PC and LS max to hear the sound. How to solve please?

Thanks!

/**
 * @file streams-i2s-webserver_wav.ino
 *
 *  This sketch reads sound data from I2S. The result is provided as WAV stream which can be listened to in a Web Browser
 *
 * @author Phil Schatzmann
 * @copyright GPLv3
 */

#include "AudioTools.h"

//AudioEncodedServer server(new WAVEncoder(),"ssid","password");  
AudioWAVServer server("myssid","mypwd"); // the same a above

I2SStream i2sStream;    // Access I2S as stream
ConverterFillLeftAndRight<int16_t> filler(LeftIsEmpty); // fill both channels - or change to RightIsEmpty

void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // start i2s input with default configuration
  Serial.println("starting I2S...");
  auto config = i2sStream.defaultConfig(RX_MODE);
  config.i2s_format = I2S_MSB_FORMAT; // if quality is bad change to I2S_LSB_FORMAT I2S_PHILIPS_FORMAT I2S_MSB_FORMAT works in 1.6.0 core https://github.com/pschatzmann/arduino-audio-tools/issues/23
  config.sample_rate = 22050;
  config.channels = 2;
  config.bits_per_sample = 16;
  i2sStream.begin(config);
  Serial.println("I2S started");

  // start data sink
  server.begin(i2sStream, config, &filler);
}

// Arduino loop  
void loop() {
  // Handle new connections
  server.doLoop();  
}

**Comment by pschatzmann** [comment, marked as answer]
Did you also have a look at the generated signal as on the serial output as csv ? 

This gives you the information about the value range that you actually get. With this informaion you could try to scale the output with https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_converter_scaler.html

**Comment by wimverlindensa** [comment]
yes I used serial plotter and als yat to check the values.
quiet -64512 / 66048
whistle loud -2496080 / 1199104
clearly 32 bit values that change when volume changes

**Comment by pschatzmann** [comment]
This can't be right if you use config.bits_per_sample = 16; you need to output int16_t values with CsvOutput<int16_t>...

**Reply by wimverlindensa** [reply]
to check the values I used streams-i2s-serial.ino , there cfg.bits_per_sample = 32;
I thought you wanted me to test with that one to be sure the mic is working

in streams-i2s-webserver_wav I not see the values of the sample, i see only below:

starting I2S...
[I] I2SConfig.h : 89 - rx/tx mode: RX
[I] I2SConfig.h : 90 - port_no: 0
[I] I2SConfig.h : 91 - is_master: Master
[I] I2SConfig.h : 92 - sample rate: 22050
[I] I2SConfig.h : 93 - bits per sample: 16
[I] I2SConfig.h : 94 - number of channels: 2
[I] I2SConfig.h : 95 - i2s_format: I2S_MSB_FORMAT
[I] I2SConfig.h : 97 - auto_clear:0
[I] I2SConfig.h : 104 - buffer_count:10
[I] I2SConfig.h : 105 - buffer_size:512
[I] I2SConfig.h : 109 - pin_bck: 14
[I] I2SConfig.h : 110 - pin_ws: 15
[I] I2SConfig.h : 111 - pin_data: 32
I2S started
[I] CodecWAV.h : 436 - is_streamed! because length is 2147418112
.....
IP address: 192.168.0.248


 AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 121 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
....................
